### PR TITLE
[Modules] Updated ContainerService/ManagedClusters to new dependency approach

### PR DESCRIPTION
**Pull request by MrMCake**
# Description

- Updated ContainerService/ManagedClusters to new dependency approach

| Pipeline |
| - |
| [ContainerService: ManagedClusters](https://github.com/Azure/ResourceModules/actions/workflows/ms.containerservice.managedclusters.yml) |

# Type of Change

Please delete options that are not relevant.

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Update to documentation

### CODE REVIEW

File: modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep [hunk 1]
```diff
@@ -0,0 +1,153 @@
+targetScope = 'subscription'
+
+// ========== //
+// Parameters //
+// ========== //
+@description('Optional. The name of the resource group to deploy for a testing purposes.')
```
Comment (MrMCake) [inline modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep:6]:
```suggestion
@description('Optional. The name of the resource group to deploy for testing purposes.')
```

File: modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        osDiskSizeGB: 0
+        osType: 'Linux'
+        serviceCidr: ''
+        storageProfile: 'ManagedDisks'
+        type: 'VirtualMachineScaleSets'
+        vmSize: 'Standard_DS2_v2'
+        vnetSubnetID: resourceGroupResources.outputs.subnetResourceIds[0]
+      }
+    ]
+    // Non-required parameters
```
Comment (eriqua) [inline modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep:82]: Pleaseremove comment
```suggestion
```

File: modules/Microsoft.ContainerService/managedClusters/.test/kubenet/deploy.test.bicep [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+        name: 'systempool'
+        osDiskSizeGB: 0
+        osType: 'Linux'
+        serviceCidr: ''
+        storageProfile: 'ManagedDisks'
+        type: 'VirtualMachineScaleSets'
+        vmSize: 'Standard_DS2_v2'
+      }
+    ]
+    // Non-required parameters
```
Comment (eriqua) [inline modules/Microsoft.ContainerService/managedClusters/.test/kubenet/deploy.test.bicep:78]: Please remove comment
```suggestion
```

File: modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(80)
+param resourceGroupName string = 'ms.containerservice.managedclusters-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'csma'
```
Comment (eriqua) [inline modules/Microsoft.ContainerService/managedClusters/.test/azure/deploy.test.bicep:14]: What about 
```suggestion
param serviceShort string = 'csmazure'
```

File: modules/Microsoft.ContainerService/managedClusters/.test/kubenet/deploy.test.bicep [hunk 5]
```diff
@@ -0,0 +1,147 @@
...
+// ========== //
+@description('Optional. The name of the resource group to deploy for testing purposes.')
+@maxLength(80)
+param resourceGroupName string = 'ms.containerservice.managedclusters-${serviceShort}-rg'
+
+@description('Optional. The location to deploy resources to.')
+param location string = deployment().location
+
+@description('Optional. A short identifier for the kind of deployment. Should be kept short to not run into resource-name length-constraints.')
+param serviceShort string = 'csmk'
```
Comment (eriqua) [inline modules/Microsoft.ContainerService/managedClusters/.test/kubenet/deploy.test.bicep:14]: What about
```suggestion
param serviceShort string = 'csmkube'
```

**Review by eriqua — APPROVED** [review]
🆗

**Pull request merged**